Enterprise-CMCS/managed-care-review
# MR-3573: Download all contracts button does not generate zip file on first click

## Summary
[MR-3573](https://qmacbis.atlassian.net/browse/MR-3573)
[Figma](https://www.figma.com/file/frKNnm6lkcpdjJ2eVJE8Bj/Managed-Care-Review?type=design&node-id=6301-158061&mode=design&t=mozBnfwnVqcVipVv-0)

This bug occurs when the `Download all` document button link is an empty string. There were two ways to reproduce this bug.
- Clicking on `Download all` before generating the bulk download URL  was finished, and the link
- Generating bulk download URL encountered an error.
Both ways caused this bug because the code defaults the URL to an empty string. This empty link will then open a new tab with the submission summary page.

The fix was to add error handling and loading UI around generating the bulk URL.
- When first loading the submission summary, `Contract details` and `Rate details` section will first show a `Loading` button. Once the bulk download URL has been generated, it will show the `Download all` button.
- If there is an error in `getBulkDlURL`, then an inline warning will appear in place of the `Download all` button, and a warning banner will appear at the top of the page.
   - There will only be one document warning banner if both Contract and Rate bulk download encountered an error.
- Fixed a few other issues:
   - Flickering on `ActionButton` component when displaying loading first.
   - `ActionButton` loading state size is larger then default size.
   - Preventing `fetchZipUrl` from executing on `DRAFT` and previous submission view.


#### Related issues

#### Screenshots

#### Test cases covered

- `DownloadButton.test.tsx`
   - `'renders loading button'`

- `ContractDetailsSummarySection.test.tsx`, `RateDetailsSummarySection.test.tsx`
   - `'renders inline error when bulk URL is unavailable'`

- `SubmissionSummary.test.tsx`
   - `'renders document download warning banner'`



## QA guidance
- You can test these error UI on the submissions created by Cypress. The submission with Q&A will have an invalid s3 url for a contract document which leads to errors when running `getBulkDlURL`. 
- Please test to make sure there is no regression in the fix for spam clicking the `Continue` or `Save as Draft` buttons.
